policy/modules/services/haproxy.te

```diff
@@ -0,0 +1,121 @@
+policy_module(haproxy)
+
+########################################
+#
+# Declarations
+#
+
+## <desc>
+##	<p>
+##	Determine whether haproxy can bind to
+##	all TCP ports.
+##	</p>
+## </desc>
+gen_tunable(haproxy_bind_all_tcp_ports, false)
+
+## <desc>
+##	<p>
+##	Determine whether haproxy can bind to
+##	kubernetes ports (typically 6443/tcp).
+##	</p>
+## </desc>
+gen_tunable(haproxy_bind_kubernetes_port, false)
+
+## <desc>
+##	<p>
+##	Determine whether haproxy can connect to
+##	all TCP ports.
+##	</p>
+## </desc>
+gen_tunable(haproxy_connect_all_tcp_ports, false)
+
+## <desc>
+##	<p>
+##	Determine whether haproxy can connect to
+##	kubernetes ports (typically 6443/tcp).
+##	</p>
+## </desc>
+gen_tunable(haproxy_connect_kubernetes_port, false)
+
+type haproxy_t;
+type haproxy_exec_t;
+init_daemon_domain(haproxy_t, haproxy_exec_t)
+
+type haproxy_conf_t;
+files_config_file(haproxy_conf_t)
+
+type haproxy_initrc_exec_t;
+init_script_file(haproxy_initrc_exec_t)
+
+type haproxy_log_t;
+logging_log_file(haproxy_log_t)
+
+type haproxy_runtime_t;
+files_runtime_file(haproxy_runtime_t)
+
+type haproxy_tmpfs_t;
+files_tmpfs_file(haproxy_tmpfs_t)
+
+########################################
+#
+# haproxy local policy
+#
+
+allow haproxy_t self:process { getsched setrlimit signal };
+allow haproxy_t self:capability { kill net_admin net_bind_service setuid setgid };
```

Comment:
The corenet bind interfaces should be providing `net_bind_service` as needed, so I'd prefer not to have it as an explicit allow. If it's not happening, then there is a bug in corenet generation scripts.

Reply:
Neat. Frankly I never knew these interfaces implied that access. I tested without `net_bind_service` explicitly given and it still has that access, so the interfaces seem to be working as intended.